arboretum/kubernetes/fetchers/fetch_cluster_resource.py

```diff
@@ -0,0 +1,159 @@
+                )
+            ]
+        )
+        cls.logger = cls.locker.logger.getChild(
+            'kubernetes.cluster_resource_fetcher'
+        )
+        return cls
+
+    def get_cluster_resource(self, server_url, token, verify, resource_types):
+        """Get resource from cluster.
+
+        :param str server_url: URL string for kubernetes API server
+        :param str token: Bearer token
+        :param str verify: Path for CA certificate file, or None if skip TLS
+                           server certificate verification (same as "curl -k")
+        :param list(str) resource_types: list of resource type in plural form
+                                         (e.g., "pods") or custom resource API
+                                         name (e.g.,
+                                         "apigroup.example.com/v1/mycustom")
+        """
+        s = self.session(server_url)
+        s.headers.update({'Authorization': f'Bearer {token}'})
+        resources = {}
+        for resource_type in resource_types:
+            if '/' in resource_type:
+                url = f'{server_url}/apis/{resource_type}'
+            else:
+                url = f'{server_url}/api/v1/{resource_type}'
+            resp = s.get(url, verify=verify)
+            if resp.status_code == 404:
+                self.logger.warning(
+                    'Cluster "%s" does not provide resource type "%s"',
+                    server_url,
+                    resource_type
+                )
+                continue
+            resp.raise_for_status()
+            resources[resource_type] = resp.json()['items']
+        return resources
+
+    @store_raw_evidence('kubernetes/cluster_resource.json')
+    def fetch_cluster_resource(self):
+        """Fetch cluster resources of listed clusters."""
+        ev = get_evidence_dependency(
+            'raw/auditree/compliance_config.json', self.locker
+        )
+        config = json.loads(ev.content)['org']['kubernetes']
+
+        cluster_list = config['cluster_list']
+        self.logger.debug('number of clusters: %d', len(cluster_list))
+
+        resource_types = self.RESOURCE_TYPE_DEFAULT
+        if ('cluster_resource' in config
+                and 'target_resource_types' in config['cluster_resource']):
+            resource_types = (
+                config['cluster_resource']['target_resource_types']
+            )
+        resources = {}
+
+        for c in cluster_list:
+            self.logger.info('Getting resource from cluster: %s', c['name'])
+            try:
+                with open(c['kubeconfig']) as f:
+                    kubeconfig = yaml.safe_load(f)
+            except OSError:
+                self.logger.error(
+                    'Cannot access cluster "%s" because '
+                    'kubeconfig file is not found: %s',
+                    c['serverURL'],
+                    c['kubeconfig']
+                )
+                raise
```

Comment:
Again, I don't see the point to this logging especially if you're going to re-raise the exception anyway.

```suggestion
        for c in cluster_list:
            with open(c['kubeconfig']) as f:
                kubeconfig = yaml.safe_load(f)
```